### PR TITLE
docs: fix mistakes in the 'Confirmation' guide

### LINKED ISSUE #771: Fix the AshAuthentication Confirmation tutorial
The tutorial uses URI.encode_query for the query parameters, but this doesn't work. Opening the link from the email provides the AuthController with the following parameters:
```elixir
%{"confirm=eyJhbGciOi....." => ""}
# should be
%{"confirm" => "eyJhbGciOi....."}
```
This can be fixed by providing the route with a keyword list instead:
```elixir
url(~p"/auth/user/confirm_new_user?#{[confirm: token]}")
```

Secondly, the tutorial uses `token: token` instead of `confirm: token` for the change confirmation part, which is incorrect.

I propose that the tutorial gets brought more into line with the [Reset Password](https://ash-hq.org/docs/guides/ash_authentication_phoenix/latest/tutorials/getting-started-with-ash-authentication-phoenix#reset-password) section of AshAuthenticationPhoenix.

<details>
<summary>new user confirmation</summary>

```elixir
defmodule Example.Accounts.User.Senders.SendNewUserConfirmationEmail do
  use AshAuthentication.Sender
  use ExampleWeb, :verified_routes

  @impl AshAuthentication.Sender
  def send(user, token, _opts) do
    Example.Accounts.Emails.deliver_email_confirmation_instructions(
      user,
      url(~p"/auth/user/confirm_new_user?#{[confirm: token]}")
    )
  end
end

defmodule Example.Accounts.Emails do
  # ...

  def deliver_email_confirmation_instructions(user, url) do
    if !url do
      raise "Cannot deliver confirmation instructions without a url"
    end

    deliver(user.email, "Confirm your email address", """
      <p>
        Hi #{user.email},
      </p>

      <p>
        Someone has tried to register a new account using this email address.
        If it was you, then please click the link below to confirm your identity.  If you did not initiate this request then please ignore this email.
      </p>

      <p>
        <a href="#{url}">Click here to confirm your account</a>
      </p>
    """)
  end

  # ...
end
```

</details>

<details>
<summary>changed attribute confirmation</summary>

```elixir
defmodule Example.Accounts.User.Senders.SendEmailChangeConfirmationEmail do
  use AshAuthentication.Sender
  use ExampleWeb, :verified_routes

  @impl AshAuthentication.Sender
  def send(user, token, _opts) do
    Example.Accounts.Emails.deliver_email_change_confirmation_instructions(
      user,
      url(~p"/auth/user/confirm_change?#{[confirm: token]}")
    )
  end
end


defmodule Example.Accounts.Emails do
  # ...

  def deliver_email_change_confirmation_instructions(user, url) do
    if !url do
      raise "Cannot deliver confirmation instructions without a url"
    end

    deliver(user.email, "Confirm your new email address", """
      <p>
        Hi #{user.email},
      </p>

      <p>
        You recently changed your email address.  Please confirm it.
      </p>

      <p>
        <a href="#{url}">Click here to confirm your new email address</a>
      </p>
    """)
  end

  # ...
end
```

</details>

### PR DESCRIPTION
**Pull request by sylvesterroos**
Closes #771 

Two main changes:
1. Fix the bugs related to encoding the tokens in the URLs.
2. Bring the modules of the guide in line with the Reset Password section of AshAuthenticationPhoenix, the style of which I personally like more.

Number 2 obviously isn't a mistake, but this guide is a bit out-of-sync if you follow the Getting Started guide from AshAuthentication and continue with the AshAuthenticationPhoenix guide.

### CODE REVIEW

File: documentation/tutorials/confirmation.md [hunk 1]
```diff
@@ -50,42 +50,86 @@ defmodule MyApp.Accounts.User do
...
+  end
+
+  # For simplicity, this module simply logs messages to the terminal.
+  # You should replace it by a proper email or notification tool, such as:
+  #
+  #   * Swoosh - https://hexdocs.pm/swoosh
+  #   * Bamboo - https://hexdocs.pm/bamboo
+  #
+  defp deliver(to, subject, body) do
+    IO.puts("Sending email to #{to} with subject #{subject} and body #{body}")
```
Comment (zachdaniel) [inline documentation/tutorials/confirmation.md:124]: Do we want this `IO.puts` here?

Comment (sylvesterroos) [reply documentation/tutorials/confirmation.md:124]: It didn't bother me per se. The line would probably get deleted anyway when you adapt the code to a production environment.

Comment (zachdaniel) [reply documentation/tutorials/confirmation.md:124]: true true. It probably helps with initial development if we recommend it.

**Comment by zachdaniel** [comment]
🚀 Thank you for your contribution! 🚀

**Comment by sylvesterroos** [comment]
Happy to help!

**Pull request merged**